httpcache_test.go

```diff
@@ -595,6 +605,54 @@ func TestGetWithDoubleVary(t *testing.T) {
 	}
 }
 
+func TestGetWithEvictOnError(t *testing.T) {
+	resetTest()
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Millisecond)
+	defer cancel()
+	req, err := http.NewRequestWithContext(ctx, "GET", s.server.URL+"/evictonerror", nil)
```

Comment:
It doesn't look like these tests ran in CI 🤔 


one change - We added a handler for `/keeponerror` but we are using `s.server.URL+"/evictonerror"` here. Need to keep both same.

Other than these the PR is looking good